### PR TITLE
[IGA] Support Condition Tests

### PR DESCRIPTION
**Pull request by rickyaristio**
**Description**
This PR adds simple single-patch tests for support penalty and Lagrange conditions.

Note: Support Nitsche implementation and test will be added soon in separate PR.

### CODE REVIEW

File: applications/IgaApplication/tests/support_condition_tests/single_patch_refined_test/single_patch_refined_test_lagrange_project_parameters.json [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                "interval": [0, "End"]
+            }
+        }],
+        "json_check_process": [{
+            "python_module": "from_json_check_result_process",
+            "kratos_module": "KratosMultiphysics",
+            "Parameters": {
+                "check_variables": ["DISPLACEMENT"],
+                "input_file_name": "support_condition_tests/single_patch_refined_test/single_patch_refined_test_lagrange_result.json",
+                "model_part_name": "IgaModelPart.StructuralAnalysis",
```
Comment (tteschemacher) [inline applications/IgaApplication/tests/support_condition_tests/single_patch_refined_test/single_patch_refined_test_lagrange_project_parameters.json]: Can you make the file comparisons only on the IgaModelPart.Load_3 model part. Those are the relevant deflections. Accordingly the result files can be much smaller.

File: applications/IgaApplication/tests/support_condition_tests/single_patch_refined_test/single_patch_refined_test_penalty_project_parameters.json [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                "interval": [0, "End"]
+            }
+        }],
+        "json_check_process": [{
+            "python_module": "from_json_check_result_process",
+            "kratos_module": "KratosMultiphysics",
+            "Parameters": {
+                "check_variables": ["DISPLACEMENT"],
+                "input_file_name": "support_condition_tests/single_patch_refined_test/single_patch_refined_test_penalty_result.json",
+                "model_part_name": "IgaModelPart.StructuralAnalysis",
```
Comment (tteschemacher) [inline applications/IgaApplication/tests/support_condition_tests/single_patch_refined_test/single_patch_refined_test_penalty_project_parameters.json]: same here.

**Review by tteschemacher — COMMENTED** [review]
Hey @rickyaristio ,

thank you very much for this implementation. Generally, it looks really good.

Please reduce the result files with the suggested change. 

Also maybe you can wrap the tests within an additional folder:
tests/**weak_support_tests**/single_patch_refined_test/...

File: applications/IgaApplication/tests/support_condition_tests/single_patch_refined_test/single_patch_refined_test_lagrange_project_parameters.json (hunk 1 above)
Comment (rickyaristio) [reply applications/IgaApplication/tests/support_condition_tests/single_patch_refined_test/single_patch_refined_test_lagrange_project_parameters.json]: Sure, thanks for the suggestion.

**Comment by tteschemacher** [comment]
now it looks great, thanks a lot for the tests.

**Review by tteschemacher — APPROVED** [review]

**Pull request merged**